shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -100,6 +100,69 @@ - (void)addInternalPlugins;
 - (void)deregisterNotifications;
 @end
 
+@interface KeyboardAnimationView : UIView
+@property(nonatomic) fml::TimePoint startTime;
+@property(nonatomic) NSTimeInterval duration;
+@property(nonatomic) CGFloat from;
+@property(nonatomic) CGFloat to;
```

Comment:
:) I like that

Reply:
Also, maybe better to use `CFTimeInterval ` or `NSTimeInterval `  for  `startTime`, because `fml::TimePoint` is a class instead of basic type like `double` or `int` ^_^

Reply:
I used `TimePoint` for easy comparison against the `FrameTimingsRecorder` - not sure if that brings in any real advantages, but thought it might help keep the time delta as accurate as possible

```objc
 fml::TimeDelta timeElapsed = recorder.get()->GetVsyncTargetTime() - keyboardView.startTime;
````

Reply:
I ignore something before, ignore that comment in https://github.com/flutter/engine/pull/37604/files#r1036653004
Sorry about that. (*/ω＼*)

Reply:
Haha no worries :)